### PR TITLE
[Rule Tuning] Use cidrMatch for eql rules checking multiple IPs

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
None

## Summary
EQL currently does not support comparing `ip` fields using `in`, so all EQL rules checking for multiple IP addresses using the convention `*.ip in ( ... )` must be converted to use `cidrMatch()` (or individual comparisons).

### CODE REVIEW

File: rules/linux/lateral_movement_telnet_network_activity_external.toml [hunk 1]
```diff
@@ -34,7 +34,8 @@ query = '''
 sequence by process.entity_id
   [process where process.name == "telnet" and event.type == "start"]
   [network where process.name == "telnet" and
-    destination.ip not in ("127.0.0.0/8", "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10", "::1/128")]
+    not cidrMatch(destination.ip, "127.0.0.0/8", "10.0.0.0/8", "172.16.0.0/12",
```
Comment (rw-access) [inline rules/linux/lateral_movement_telnet_network_activity_external.toml:37]: so we have a few options here
* add ipv6 to cidrMatch for EQL. this is a big ask unfortunately
* use `cidrMatch` but monkey patch the existing EQL function, just to pass unit tests
* rename to `cidrmatch` and use `with eql.parser.ignore_unknown_functions:` when parsing

Comment (brokensound77) [reply rules/linux/lateral_movement_telnet_network_activity_external.toml:37]: lets go with option 3 for this and aim for option 1 for 7.11

Updating now

File: rules/windows/command_and_control_certutil_network_connection.toml [hunk 2]
```diff
@@ -26,7 +26,7 @@ query = '''
 sequence by process.entity_id
   [process where process.name : "certutil.exe" and event.type == "start"]
   [network where process.name : "certutil.exe" and
-    destination.ip not in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
+    cidrMatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```
Comment (rw-access) [inline rules/windows/command_and_control_certutil_network_connection.toml:29]: where'd the `not` go?

File: rules/windows/defense_evasion_misc_lolbin_connecting_to_the_internet.toml [hunk 3]
```diff
@@ -30,7 +30,7 @@ sequence by process.entity_id
      event.type == "start"]
   [network where (process.name : "expand.exe" or process.name : "extrac.exe" or
                  process.name : "ieexec.exe" or process.name : "makecab.exe") and
-     destination.ip not in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
+     cidrMatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```
Comment (rw-access) [inline rules/windows/defense_evasion_misc_lolbin_connecting_to_the_internet.toml:33]: lost the `not`

File: rules/windows/command_and_control_certutil_network_connection.toml [hunk 4]
```diff
@@ -26,7 +26,7 @@ query = '''
 sequence by process.entity_id
   [process where process.name : "certutil.exe" and event.type == "start"]
   [network where process.name : "certutil.exe" and
-    destination.ip not in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
+    cidrmatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```
Comment (brokensound77) [inline rules/windows/command_and_control_certutil_network_connection.toml:29]:
```suggestion
    not cidrmatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```

File: rules/windows/defense_evasion_misc_lolbin_connecting_to_the_internet.toml [hunk 5]
```diff
@@ -30,7 +30,7 @@ sequence by process.entity_id
      event.type == "start"]
   [network where (process.name : "expand.exe" or process.name : "extrac.exe" or
                  process.name : "ieexec.exe" or process.name : "makecab.exe") and
-     destination.ip not in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
+     cidrmatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_misc_lolbin_connecting_to_the_internet.toml:33]:
```suggestion
     not cidrmatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16")]
```

File: rules/linux/lateral_movement_telnet_network_activity_internal.toml [hunk 6]
```diff
@@ -34,8 +34,8 @@ query = '''
 sequence by process.entity_id
   [process where process.name == "telnet" and event.type == "start"]
   [network where process.name == "telnet" and
-    destination.ip in ("10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10") and
-    destination.ip not in ("127.0.0.0/8", "::1/128")]
+    cidrmatch(destination.ip, "10.0.0.0/8", "172.16.0.0/12", "192.168.0.0/16", "FE80::/10") and
+    not cidrmatch(destination.ip, "127.0.0.0/8", "::1/128")]
```
Comment (rw-access) [inline rules/linux/lateral_movement_telnet_network_activity_internal.toml:38]: can you or @bm11100 add one screenshot for `cidrmatch` (lowercase) working in the ES API?

**Comment by brokensound77** [comment]
No errors!
image

File: rules/linux/lateral_movement_telnet_network_activity_internal.toml (hunk 6 above)
Comment (brokensound77) [reply rules/linux/lateral_movement_telnet_network_activity_internal.toml:38]: added

**Comment by brokensound77** [comment]
with results
image

**Review by rw-access — APPROVED** [review]
➕

**Review by bm11100 — APPROVED** [review]
👏

**Pull request merged**